lib/jsonrpc_client.ml

```diff
@@ -17,49 +17,101 @@
 module D = Debug.Make(struct let name = "jsonrpc_client" end)
 open D
 
-let input_json_object fin =
-	let buf = Buffer.create 1024 in
-	let brace_cnt = ref 0 in
-	let in_string = ref false in
-	let last_char () = Buffer.nth buf (Buffer.length buf - 1) in
-	let rec get () =
-		let c = input_char fin in
+exception Timeout
+
+let json_rpc_max_len = ref 65536 (* Arbitrary maximum length of RPC response *)
+let json_rpc_read_timeout = ref "60000000000"
+let json_rpc_write_timeout = ref "60000000000"
+
+let to_s  s = (Int64.to_float s) *. 1e-9
+
+(* Read the entire contents of the fd, of unknown length *)
+let timeout_read fd timeout =
+	let buf = Buffer.create !json_rpc_max_len in
+	let rec inner max_time max_bytes =
+		let c = Mtime_clock.counter () in
+		let get_used_time counter = Mtime.Span.to_uint64_ns (Mtime_clock.count counter) in
+		let (ready_to_read, _, _) = Unix.select [fd] [] [] (to_s max_time) in
```

Comment:
I agree with you that to deal with error situation might be better, in case of fd issue; However, I searched our code base, many similar usage, without dealing the error; I also searched only, few such examples. So I guess we just stick to this style right now.